### PR TITLE
[dml] Pull out Redshift UPDATE queries

### PR DESCRIPTION
**Pull request by nathan-artie**

### CODE REVIEW

File: lib/destination/dml/merge.go [hunk 1]
```diff
@@ -98,6 +98,38 @@ func (m *MergeArgument) buildRedshiftInsertQuery(columns []columns.Column) strin
...
+
+	return fmt.Sprintf(`UPDATE %s as c SET %s FROM %s as cc WHERE %s%s;`,
+		// UPDATE table set col1 = cc. col1
+		m.TableID.FullyQualifiedName(), buildColumnsUpdateFragment(columns, m.Dialect),
+		// FROM table (temp) WHERE join on PK(s)
+		m.SubQuery, strings.Join(m.redshiftEqualitySQLParts(), " and "), idempotentClause,
+	)
+}
+
+func (m *MergeArgument) buildRedshiftUpdateQuery(columns []columns.Column) string {
```
Comment (nathan-artie) [inline lib/destination/dml/merge.go:117]: Will look at combining these two in a follow up PR.

File: lib/destination/dml/merge.go [hunk 2]
```diff
@@ -98,6 +98,38 @@ func (m *MergeArgument) buildRedshiftInsertQuery(columns []columns.Column) strin
...
+	// We also need to do staged table's idempotency key is GTE target table's idempotency key
+	// This is because Snowflake does not respect NS granularity.
+	var idempotentClause string
+	if m.IdempotentKey != "" {
+		idempotentClause = fmt.Sprintf(" AND cc.%s >= c.%s", m.IdempotentKey, m.IdempotentKey)
+	}
+
+	return fmt.Sprintf(`UPDATE %s as c SET %s FROM %s as cc WHERE %s%s AND COALESCE(cc.%s, false) = false;`,
```
Comment (nathan-artie) [inline lib/destination/dml/merge.go:125]: Just moved this, no changes.

File: lib/destination/dml/merge.go [hunk 3]
```diff
@@ -98,6 +98,38 @@ func (m *MergeArgument) buildRedshiftInsertQuery(columns []columns.Column) strin
...
 
+func (m *MergeArgument) buildRedshiftSoftDeleteUpdateQuery(columns []columns.Column) string {
+	// We also need to do staged table's idempotency key is GTE target table's idempotency key
+	// This is because Snowflake does not respect NS granularity.
+	var idempotentClause string
+	if m.IdempotentKey != "" {
+		idempotentClause = fmt.Sprintf(" AND cc.%s >= c.%s", m.IdempotentKey, m.IdempotentKey)
+	}
+
+	return fmt.Sprintf(`UPDATE %s as c SET %s FROM %s as cc WHERE %s%s;`,
```
Comment (nathan-artie) [inline lib/destination/dml/merge.go:109]: Just moved this, no changes.

**Review by Tang8330 — APPROVED** [review]

File: lib/destination/dml/merge.go (hunk 3 above)
Comment (nathan-artie) [reply lib/destination/dml/merge.go:109]: Err, I lied, I changed `m.Columns` to columns.

**Pull request merged**